packages/browser-wallet/src/popup/store/settings.ts

```diff
@@ -12,18 +13,17 @@ export const encryptedSeedPhraseAtom = atomWithChromeStorage<EncryptedData | und
 );
 export const themeAtom = atomWithChromeStorage<Theme>(ChromeStorageKey.Theme, Theme.Light);
 
-const storedJsonRpcUrlAtom = atomWithChromeStorage<string>(ChromeStorageKey.JsonRpcUrl, '', true);
-export const jsonRpcUrlAtomLoading = atom<AsyncWrapper<string>, string>(
-    (get) => get(storedJsonRpcUrlAtom),
-    (_, set, jsonRpcUrl) => {
-        set(storedJsonRpcUrlAtom, jsonRpcUrl);
-        popupMessageHandler.broadcast(EventType.ChainChanged, jsonRpcUrl);
-    }
+const storedNetworkConfigurationAtom = atomWithChromeStorage<NetworkConfiguration>(
+    ChromeStorageKey.NetworkConfiguration,
+    mainnet
 );
-export const jsonRpcUrlAtom = atom<string, string>(
-    (get) => get(jsonRpcUrlAtomLoading).value,
-    (_, set, jsonRpcUrl) => {
-        set(jsonRpcUrlAtomLoading, jsonRpcUrl);
+export const networkConfigurationAtom = atom<NetworkConfiguration, NetworkConfiguration>(
+    (get) => get(storedNetworkConfigurationAtom),
+    (_, set, networkConfiguration) => {
+        set(storedNetworkConfigurationAtom, networkConfiguration);
+
+        // TODO What does it make sense to broadcast with this event? It the event even needed anymore?
+        popupMessageHandler.broadcast(EventType.ChainChanged, networkConfiguration.genesisHash);
```

Comment:
I think we should keep the event. It makes sense to give the hash (it's the only identifier we have), the only other option is the name, but ehh...

Reply:
I think the hash is the most sensible. I am not sure what people would use it for, but we will keep it for now.